packages/@react-aria/button/src/useButton.ts

```diff
@@ -76,6 +76,10 @@ export function useButton(props: AriaButtonProps<ElementType>, ref: RefObject<an
       rel: elementType === 'a' ? rel : undefined
     };
   }
+  // don't apply type button to non button elements
```

Comment:
Have to do this here because additionalProps was moved earlier in the merge, so now the defaults override it instead of the other way around.
I'm not sure if that was intentional, there are likely some other things that are now misbehaving because of that move.

Reply:
We need both checks?

Reply:
Can you move this logic into ` if (elementType !== 'button') {` section above?
```
type: elementType === 'input' && type !== 'button' ? type : undefined,
```
As for `disabled` and `type` being overridden, it might be because those attributes aren't supported for every possible elementType (https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled). We would want to make sure `disabled=undefined` for elementTypes that don't fit in that list. Same for the `type`

Reply:
right, so additionalProps has the right things for elementType !== button, and lets say elementType is 'a', then additionalProps.type will be undefined
however, type has a default of button up in the destructure of props

the type from the destructure of props will override the additionalProps.type https://github.com/adobe/react-spectrum/pull/1772/files#diff-a12dc137bf98877d8d127f2ea766a00dbce207314989d88bf6719982d9e866abR106

Reply:
Ah whoops, misread the code below. Feels like additionalProps should be moved below `type` in the return then? Seems like it was moved in the first place because of https://github.com/adobe/react-spectrum/commit/a08b61b3c3000f4084065b90806fd9258e1931b1, which perhaps because we wanted the tabindex from buttonProps to override the additionProps tabindex? Not sure, haven't played around with it locally but if that is the case then perhaps the order of the final mergeProps is to shift additionalProps to be the 2nd arg and buttonProps as the third arg

Reply:
Unfortunately that's not right either, as that would be roughly

```
// for elementType === 'a'
mergeProps(
  {type: 'button'}, // from default props
  {type: undefined},
  {}
);
```
which would still yield {type: 'button'} because undefined won't erase it from the object
so I think the only way to do it is to actually set the right thing with an extra check like I have here

Reply:
urgh, I see, so thats why we had previously [spread](https://github.com/adobe/react-spectrum/commit/a08b61b3c3000f4084065b90806fd9258e1931b1#diff-a12dc137bf98877d8d127f2ea766a00dbce207314989d88bf6719982d9e866abL100) `additionalProps`...  Well a new concern that I have is that the `disabled: isDisabled` bit will suffer from a similar issue (i.e. elementType === 'a' will have the `disabled` attribute which isn't a supported attribute). Probably can fix that by adding another if statement similar to your `type` one or by doing:
```
    buttonProps: mergeProps({
      'aria-haspopup': props['aria-haspopup'],
      'aria-expanded': props['aria-expanded'],
      'aria-controls': props['aria-controls'],
      'aria-pressed': props['aria-pressed'],
      disabled: isDisabled,
      type,
      ...(additionalProps || {}),
      onClick: (e) => {
        if (deprecatedOnClick) {
          deprecatedOnClick(e);
          console.warn('onClick is deprecated, please use onPress');
        }
      }
    }, buttonProps)
```
Either approach is fine w/ me, and I _think_ that should the last thing, don't see any other possible mergeProps issues (onClick should be fine I believe).